packages/strapi-utils/lib/logger.js

```diff
@@ -4,18 +4,52 @@
  * Logger.
  */
 
-const pino = require('pino');
+
+const fs = require('fs');
+const path = require('path');
 const _ = require('lodash');
+const pino = require('pino');
 
 const logLevels = ['fatal', 'error', 'warn', 'info', 'debug', 'trace'];
 
 function getLogLevel() {
-  if (_.isString(process.env.STRAPI_LOG_LEVEL) && _.includes(logLevels, process.env.STRAPI_LOG_LEVEL.toLowerCase())) {
+  if (
+    _.isString(process.env.STRAPI_LOG_LEVEL) &&
+    _.includes(logLevels, process.env.STRAPI_LOG_LEVEL.toLowerCase())
+  ) {
     return process.env.STRAPI_LOG_LEVEL;
   }
   return 'debug';
 }
 
+const logTypes = ['stdout', 'file'];
+function getLogType() {
+  if (
+    _.isString(process.env.STRAPI_LOG_TYPE) &&
+    _.includes(logTypes, process.env.STRAPI_LOG_TYPE.toLowerCase())
+  ) {
+    return process.env.STRAPI_LOG_TYPE;
+  }
+  return 'stdout';
+}
+
+function parseLogPath(defaultLogFilename = 'strapi.log') {
+  if (_.isString(process.env.STRAPI_LOG_PATH)) {
+    return path.parse(process.env.STRAPI_LOG_PATH);
+  }
+  return path.parse(path.join(__dirname, defaultLogFilename));
+}
+
+function getLogDir() {
+  return parseLogPath().dir;
+}
+
+function getLogBase() {
+  return parseLogPath().base;
+}
+
+const logFile = path.join(getLogDir(), getLogBase());
```

Comment:
I have written code with more features than the present one which is a fork and this split/join is currently useless.

In my case I appreciate this kind of optional and basic feature because I don t use strapi in production but I use it like a toolbox to build very quick POCs of API based WebApp without any devops tools and I need to have  logs because I want to save them just with envt setting Of course it s not a question of life but of less effort because I have to be quick.

Finally it's up to strapi team to decide.

Reply:
> this current commit was only the first US/step before to go further in this feature 

Exactly my point! Introducing this meaning support more code, more features, more work that 1) is not the scope of what Strapi is; and 2) there are specialized tools for that. Whatever that we are trying to archive with this is already solve in a better way by others.

And, is just my opinion, please don't drop anything, I'm just trying to improve this project, just like you 😄

Reply:
Just this current PR of course :relaxed:
Thx.